src/signing.rs

```diff
@@ -153,6 +153,7 @@ impl SigningKey {
         self.verifying_key
     }
 
+    #[cfg(any(test, feature = "rand_core"))]
```

Comment:
The gate is already set here for generate() and it ensures we don't need to gate doctests in teh doc itself
We don't gate with docsrs since we setup the features for docsrs separately in the docs PR
This aligns with everything else - the gates sit above comments always.

Reply:
> the gates sit above comments always.

FWIW I've found that style to be... unusual to me.

I'm used to a "comments, attributes, code" style which is used by e.g. rustc

Reply:
It's easier to read imo and we have it elsewhere like that ? we could always change that

Reply:
It's also that we have some attr's middle of doctests.. I'm pondering about the doctests

Reply:
My vote would be to change it as I haven't seen it used anywhere else. 

Unfortunately I don't see the "comments, attributes, code" ordering actually prescribed in any style guide, but it's pretty consistently that way in every project I've seen maintained by the Rust team

Reply:
Your wish shall be granted. psst we also needed it for test as there are some test fn's that use generate internally.

Reply:
fwiw - docsrs is / has been running nightly and we would only set the unstable flag with docsrs anyways ?
`#![cfg_attr(docsrs, feature(doc_auto_cfg))]`
We already use doc_auto_cfg in curve25519
https://github.com/dalek-cryptography/curve25519-dalek/blob/main/src/lib.rs#L14

Reply:
I've migrated several @RustCrypto projects to using it, and it seems to be working fine to me. I've checked the output versus the manual annotations and haven't found any discrepancies besides things I missed in manual annotations

Reply:
Yah, I took the manual annotation out. Thanks for picking this.